stubs/docker/docker/client.pyi

```diff
@@ -12,11 +13,25 @@ from docker.models.services import ServiceCollection
 from docker.models.swarm import Swarm
 from docker.models.volumes import VolumeCollection
 
+@type_check_only
+class _Environ(Protocol):
+    def __getitem__(self, k: str, /) -> str: ...
+    def keys(self) -> Iterable[str]: ...
+
 class DockerClient:
     api: APIClient
     def __init__(self, *args, **kwargs) -> None: ...
     @classmethod
-    def from_env(cls, **kwargs) -> DockerClient: ...
+    def from_env(
+        cls,
+        *,
+        version: str | None = None,
+        timeout: int | None = None,
+        max_pool_size: int | None = None,
+        environment: _Environ | None = None,
+        credstore_env: dict[str, str] | None = None,
+        use_ssh_client: bool | None = None,
```

Comment:
Looking at `APIClient`, some of these don't take `None`. Also, while `credstore_env` is mentioned in the docstring, I can't find it anywhere in the source, and using it causes a runtime error:

```
>>> from docker.client import DockerClient
>>> DockerClient.from_env(credstore_env={})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/srittau/foo/lib/python3.12/site-packages/docker/client.py", line 99, in from_env
    **kwargs_from_env(**kwargs)
      ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: kwargs_from_env() got an unexpected keyword argument 'credstore_env'
```

```suggestion
        version: str | None = None,
        timeout: int = ...,
        max_pool_size: int = ...,
        environment: _Environ | None = None,
        use_ssh_client: bool = False,
```

Reply:
Great digging! I'll try to do similar in follow-ups for other methods now I further understand the process.